lib/httparty.rb

```diff
@@ -566,9 +566,19 @@ def perform_request(http_method, path, options, &block) #:nodoc:
     def process_headers(options)
       if options[:headers] && headers.any?
         options[:headers] = headers.merge(options[:headers])
+        options[:headers] = process_dynamic_headers(options[:headers])
       end
     end
 
+    def process_dynamic_headers(headers)
+      headers = headers.map do |header, value|
```

Comment:
What you want here is [Enumerable#each_with_object](https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-each_with_object). Resulting code will much cleaner, than map + conversion to hash

Reply:
@TheSmartnik thanks for the suggestion, changed the code accordingly.